src/bin/tools/unzip.ts

```diff
@@ -1,184 +1,89 @@
-    const dirsCreated: (string | undefined)[] = [];
-    dirsCreated.push(await mkdir(extractDirPath, { recursive: true }));
-    const promises: Promise<string>[] = [];
-
-    // Iterate over all files in the zip, skip files which are not in archiveDir,
-    // if given.
-    for await (const record of iterateZipArchive(zipFilePath)) {
-        const { path: recordPath, createReadStream: createRecordReadStream } = record;
-        if (pathOfDirToExtractInArchive && !recordPath.startsWith(pathOfDirToExtractInArchive)) {
-            continue;
+import fsp from "node:fs/promises";
+import fs from "fs";
+import path from "node:path";
+import yauzl from "yauzl";
+import stream from "node:stream";
+import { promisify } from "node:util";
+
+const pipeline = promisify(stream.pipeline);
+
+async function pathExists(path: string) {
+    try {
+        await fsp.stat(path);
+        return true;
+    } catch (error) {
+        if ((error as { code: string }).code === "ENOENT") {
+            return false;
         }
-        const relativePath = pathOfDirToExtractInArchive ? pathRelative(pathOfDirToExtractInArchive, recordPath) : recordPath;
-        const filePath = pathJoin(extractDirPath, relativePath);
-        const parent = pathDirname(filePath);
-        promises.push(
-            new Promise<string>(async (resolve, reject) => {
-                if (!dirsCreated.includes(parent)) dirsCreated.push(await mkdir(parent, { recursive: true }));
-
-                // Pull the file out of the archive, write it to the target directory
-                const output = createWriteStream(filePath);
-                output.on("error", e => reject(Object.assign(e, { filePath })));
-                output.on("finish", () => resolve(filePath));
-                createRecordReadStream().pipe(output);
-            })
-        );
+        throw error;
     }
+}
 
-    // Wait until _all_ files are either extracted or failed
-    const [success, failure] = (await Promise.allSettled(promises)).reduce(...partitionPromiseSettledResults<string>());
-
-    // If any extraction failed, try to clean up, then throw a MultiError,
-    // which has a `cause` field, containing a list of root cause errors.
-    if (failure.length) {
-        await Promise.all([
-            ...success.map(path => unlink(path).catch(_unused => undefined)),
-            ...failure.map(e => e && e.path && unlink(e.path as string).catch(_unused => undefined))
-        ]);
-        await Promise.all(dirsCreated.filter(Boolean).sort(sortByFolderDepth("desc")));
-        const e = new Error("Failed to extract: " + failure.map(e => e.message).join(";"));
-        (e as any).cause = failure;
-        throw e;
+export async function unzip(file: string, targetFolder: string, unzipSubPath?: string) {
+    // add trailing slash to unzipSubPath and targetFolder
+    if (unzipSubPath && (!unzipSubPath.endsWith("/") || !unzipSubPath.endsWith("\\"))) {
+        unzipSubPath += "/";
```

Comment:
maybe add a comment that here we must not use path.sep, cause in a zipfile, _only_ forward slashes are valid path separators?